File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/twosquare/TwoSquareKey.kt
package cryptobench.ciphers.twosquare

import cryptobench.ciphers.playfair.PlayfairKey

data class TwoSquareKey(val topWord: String, val bottomWord: String) {

    fun topSquare(): String = PlayfairKey(topWord).square()

    fun bottomSquare(): String = PlayfairKey(bottomWord).square()

    companion object {
        fun default(): TwoSquareKey = TwoSquareKey("EXAMPLE", "KEYWORD")
    }
}

File: fixtures/scale/dart/large/lib/ciphers/twosquare/twosquare_key.dart
import '../playfair/playfair_key.dart';

class TwoSquareKey {
  final String topWord;
  final String bottomWord;

  TwoSquareKey(this.topWord, this.bottomWord);

  String topSquare() => PlayfairKey(topWord).square();

  String bottomSquare() => PlayfairKey(bottomWord).square();

  factory TwoSquareKey.defaultKey() => TwoSquareKey('EXAMPLE', 'KEYWORD');
}
